src/network.py

```diff
@@ -178,17 +178,12 @@ def __init__(self, d_in, hidden, bidirectional=False, num_layers=1, dropout=.2,
         self.rnn = RNNEncoder(d_in, hidden, bidirectional=bidirectional, num_layers=num_layers, dropout=dropout)
         self.dropout = nn.Dropout(p=dropout)
         self.non_linear = nn.LeakyReLU(relu)
-        self.reduce_h_W = nn.Linear(hidden * self.num_dir, hidden)
         self.fc2 = nn.Linear(hidden, 2)
 
     def forward(self, out, out_len):
         _, (e_h, _) = self.rnn(out, out_len)
-        if self.num_dir == 2:
-            h = e_h.view(self.num_layers, self.num_dir, -1, self.hidden)
-            out = torch.cat((h[:, 0, :, :], h[:, 1, :, :]), dim=-1)
-        else:
-            out = e_h[-1]
-        return self.fc2(self.dropout(self.non_linear(self.reduce_h_W(out))))
+        # -1 gets topmost layer I think
+        return self.fc2(self.dropout(self.non_linear(e_h[-1])))
```

Comment:
The output is "h_n of shape (num_layers * num_directions, batch, hidden_size)" according to https://pytorch.org/docs/stable/generated/torch.nn.LSTM.html#torch.nn.LSTM. I don't think -1 can just do this though, but I should test it first.

Reply:
Nevermind, I thought RNN was an `LSTM`, but it turns out to be an `RNNEncoder`.